File: 06ros/ws02/second_ws/src/use_cus2_srv_pkg/scripts/client_cus2_srv.py
#! /usr/bin/env python
# coding=utf8

import rospy
from custom2_srvs.srv import GenStudentId, GenStudentIdRequest, GenStudentIdResponse

if __name__ == '__main__':
    nodeName = "client_node"
    rospy.init_node(nodeName)

    serviceName = "/zcb01/service"
    client = rospy.ServiceProxy(serviceName, GenStudentId)
    client.wait_for_service()

    request = GenStudentIdRequest()
    request.stu.name = "tom"
    request.stu.age = 18

    try:
        response = client.call(request)
        if isinstance(response, GenStudentIdResponse):
            rospy.loginfo("call success,result is %s" % response.id)
    except Exception as e:
        print e
        rospy.loginfo("call failed")

File: 06ros/ws02/second_ws/src/use_cus2_srv_pkg/src/client_cus2_srv.cpp
//
// Created by zcb on 5/4/21.
//

#include <ros/ros.h>
#include <iostream>
#include <custom2_srvs/GenStudentId.h>


int main(int argc, char **argv) {

    std::string nodeName = "client_node";
    ros::init(argc, argv, nodeName);

    ros::NodeHandle node;

    std::string serviceName = "/zcb01/service";
    ros::ServiceClient client = node.serviceClient<custom2_srvs::GenStudentIdRequest, custom2_srvs::GenStudentIdResponse>(
            serviceName);
    client.waitForExistence();


    custom2_srvs::GenStudentId service;
    service.request.stu.name = "tom";
    service.request.stu.age = 18;
    if (client.call(service)) {
        ROS_INFO_STREAM("call success,res is " << service.response.id);
    } else {
        ROS_INFO_STREAM("call failed!");
    }


    return 0;
}





